### DISCUSSION TITLE
centos6.4 启动失败

### DISCUSSION THREAD

**Discussion by 7y-9** [alibaba/ilogtail#1237] [Help] [answered]
我尝试使用centos6.4运行centos7.4编译出来的产物，没有成功，又尝试运行官方发行版。也没有启动成功。显示缺少依赖，哪位大佬可以帮看看这个问题。

`[root@i-j8zdncbq ~]# wget https://ilogtail-community-edition.oss-cn-shanghai.aliyuncs.com/latest/ilogtail-latest.linux-amd64.tar.gz
--2023-11-14 09:53:54--  https://ilogtail-community-edition.oss-cn-shanghai.aliyuncs.com/latest/ilogtail-latest.linux-amd64.tar.gz
Resolving ilogtail-community-edition.oss-cn-shanghai.aliyuncs.com... 106.14.228.175
Connecting to ilogtail-community-edition.oss-cn-shanghai.aliyuncs.com|106.14.228.175|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 44496979 (42M) [application/x-gzip]
Saving to: “ilogtail-latest.linux-amd64.tar.gz”

100%[============================================>] 44,496,979  61.1M/s   in 0.7s    

2023-11-14 09:53:57 (61.1 MB/s) - “ilogtail-latest.linux-amd64.tar.gz” saved [44496979/44496979]

[root@i-j8zdncbq ~]# tar -xzvf ilogtail-latest.linux-amd64.tar.gz
ilogtail-1.8.1/
ilogtail-1.8.1/libPluginAdapter.so
ilogtail-1.8.1/user_yaml_config.d/
ilogtail-1.8.1/LICENSE
ilogtail-1.8.1/README.md
ilogtail-1.8.1/ilogtail_config.json
ilogtail-1.8.1/libebpf.so
ilogtail-1.8.1/ilogtail
ilogtail-1.8.1/libPluginBase.so
[root@i-j8zdncbq ~]# cd ilogtail-
ilogtail-1.8.1/                     ilogtail-latest.linux-amd64.tar.gz
[root@i-j8zdncbq ~]# cd ilogtail-
ilogtail-1.8.1/                     ilogtail-latest.linux-amd64.tar.gz
[root@i-j8zdncbq ~]# cd ilogtail-1.8.1/
[root@i-j8zdncbq ilogtail-1.8.1]# ls
ilogtail              libebpf.so           libPluginBase.so  README.md
ilogtail_config.json  libPluginAdapter.so  LICENSE           user_yaml_config.d
[root@i-j8zdncbq ilogtail-1.8.1]# ./ilogtail &
[1] 763502
[root@i-j8zdncbq ilogtail-1.8.1]# ./ilogtail: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.18' not found (required by ./ilogtail)
./ilogtail: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.14' not found (required by ./ilogtail)
./ilogtail: /usr/lib64/libstdc++.so.6: version `CXXABI_1.3.5' not found (required by ./ilogtail)
./ilogtail: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.15' not found (required by ./ilogtail)
./ilogtail: /usr/lib64/libstdc++.so.6: version `GLIBCXX_3.4.19' not found (required by ./ilogtail)
./ilogtail: /lib64/libc.so.6: version `GLIBC_2.17' not found (required by ./ilogtail)

[1]+  Exit 1                  ./ilogtail
[root@i-j8zdncbq ilogtail-1.8.1]# 
[root@i-j8zdncbq ilogtail-1.8.1]# 
[root@i-j8zdncbq ilogtail-1.8.1]# uname -r
2.6.32-358.el6.x86_64
[root@i-j8zdncbq ~]# cat /etc/centos-release
CentOS release 6.4 (Final)
`

**Comment by yyuuttaaoo** [comment]
你可能需要对ilogtail进行重新编译，打开ENABLE_COMPATIBLE_MODE编译开关，可以手动改一下core/CMakeList.txt

**Reply by 7y-9** [reply]
我试一下

**Reply by 7y-9** [reply]
还是失败 我将core/CMakeLists.txt中的这个开关
option(ENABLE_COMPATIBLE_MODE "Build Logtail in compatible mode (for low version Linux)" ON)
直接设置为ON，用7.4来编译，得到的产物在centos6.4上运行 还是会报错
<img width="1088" alt="image" src="https://github.com/alibaba/ilogtail/assets/121850020/e6dc079a-e59c-4558-a06c-190f02f0d10e">

**Reply by 7y-9** [reply, marked as answer]
在 scripts/gen_build_scripts.sh 把这两个值默认改为on后编译 在低版本可以成功运行

ENABLE_COMPATIBLE_MODE=${ENABLE_COMPATIBLE_MODE:-ON}
ENABLE_STATIC_LINK_CRT=${ENABLE_STATIC_LINK_CRT:-ON}
